tests-web/www/js/layers.js

```diff
@@ -1,7 +1,27 @@
 export const makeJsonLayer = (text) => {
-    return new L.geoJSON(JSON.parse(text), { style: { color: '#f55' }});
+    return new L.geoJSON(JSON.parse(text), { style: styleGeoJson });
 };
 
+const intersectionColours = {
+    MapEdge: '#696',
+    Connection: '#666',
+    MultiConnection: '#669',
+    Merge: '#969',
+    Crossing: '#999',
+    Terminus: '#966',
+}
+
+const styleGeoJson = (feature) => {
+    if (feature.geometry.type === 'Polygon') {
+        return {
+            color: intersectionColours[feature.properties?.complexity] || '#666',
```

Comment:
You can do `?` in the middle there and have `||` at the end? O_O   The `[]` returns null, so the map lookup returns null. Very elegant, much more readable than Rust-style `features.properties.get("complexity").and_then(|c| intersectionColors.get(c)).unwrap_or_else("#666")`

Reply:
`?.` is actually the operator, which returns `undefined` if the LHS is `null` or `undefined` (instead of crashing), otherwise acts like `.`. It is guarding against `properties` not being there (thought it probably always will)

`x[undefined]` is coerced to `x['undefined']` (because JS) which will return `undefined` in our case, so we can use `||` to chain to the default value. Chaining with `||`, `&&`, and now `??` and `???` can be quite elegant.